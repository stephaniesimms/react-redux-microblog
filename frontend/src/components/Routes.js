import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import Homepage from './Homepage';
import Post from '../containers/Post';
import NewPost from '../containers/NewPost';
import NotFound from '../components/NotFound';

/** Routes to:
 * 
 *   - homepage
 *   - new form
 *   - individual posts
 *   - 404 page not found
 */
class Routes extends Component {
  render() {
    return (
      <Switch>
        <Route path='/' exact render={() => <Homepage />} />
        <Route path='/new' exact render={rtProps => <NewPost {...rtProps}/>} />
        <Route path='/posts/:postId' exact render={rtProps => <Post {...rtProps} />} />
        <Route render={() => <NotFound />} />
      </Switch>
    );
  }
}

export default Routes;