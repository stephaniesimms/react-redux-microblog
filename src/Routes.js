import React, { Component } from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';
import Homepage from './Homepage';
import PostForm from './PostForm';
import Post from './Post';

class Routes extends Component {
  render() {
    return (
      <Switch>
        <Route path="/" exact render={() => <Homepage posts={this.props.posts}/>} />
        <Route path="/new" exact render={rtProps => <PostForm {...rtProps} addPost={this.props.addPost}/>} />
        <Route path="/:postId" exact render={rtProps => <Post {...rtProps} 
          posts={this.props.posts}
          deletePost={this.props.deletePost}/>} />
        <Redirect to="/" />
      </Switch>
    );
  }
}

export default Routes;