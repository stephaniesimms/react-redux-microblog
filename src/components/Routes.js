import React, { Component } from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';
import Homepage from './Homepage';
import NewPost from '../containers/NewPost';
import Post from '../containers/Post';

class Routes extends Component {
  render() {
    return (
      <Switch>
        <Route path="/" exact render={() => <Homepage posts={this.props.posts}/>} />
        <Route path="/new" exact render={rtProps => <NewPost {...rtProps} 
          addPost={this.props.addPost} 
          formType="New Post"/> } />
        <Route path="/:postId" exact render={rtProps => <Post {...rtProps} 
          posts={this.props.posts}
          deletePost={this.props.deletePost}
          updatePost={this.props.updatePost}
          addComment={this.props.addComment}
          deleteComment={this.props.deleteComment}
          />} />
        <Redirect to="/" />
      </Switch>
    );
  }
}

export default Routes;