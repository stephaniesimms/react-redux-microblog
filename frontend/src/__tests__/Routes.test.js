import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Routes from '../components/Routes';

it('renders without crashing', function () {
  shallow(<Routes />);
});

it('matches snapshot', function () {
  let wrapper = shallow(<Routes />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});