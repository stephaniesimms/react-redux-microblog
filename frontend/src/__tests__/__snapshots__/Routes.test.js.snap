// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/new"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/posts/:postId"
    render={[Function]}
  />
  <Route
    render={[Function]}
  />
</Switch>
`;
